Task_1_4_1/src/main/java/RecordBook.java

```diff
@@ -0,0 +1,153 @@
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Class that implements the work of the student's electronic record book.
+ */
+public class RecordBook {
+    private Integer amountOfAllGrades;
+    private Integer sumOfAllGrades;
+    private Integer amountOfThrees;
+    private HashMap<String, ArrayList<Integer>> finalGrades;
+    private HashMap<Integer, HashMap<String, Integer>> grades;
+
+    /**
+     * Constructor.
+     */
+    public RecordBook() {
+        this.amountOfAllGrades = 0;
+        this.sumOfAllGrades = 0;
+        this.amountOfThrees = 0;
+        this.grades = new HashMap<>();
+        for (int i = 0; i < 8; i++) {
+            this.grades.put(i + 1, new HashMap<>());
+        }
+        this.finalGrades = new HashMap<>();
+    }
+
+    /**
+     * Converting a grade from a string to a number.
+     *
+     * @param grade Grade as string
+     * @return Grade as number
+     */
+    private int convertGrade(String grade) {
+        if (grade.compareTo("удовлетворительно") == 0) {
+            return 3;
+        } else if (grade.compareTo("хорошо") == 0) {
+            return 4;
+        } else if (grade.compareTo("отлично") == 0) {
+            return 5;
+        } else {
+            return 0;
+        }
+    }
+
+    /**
+     * Function for putting the grade into the record book.
+     * If the grade for a particular subject in the semester has already been set, then it can be changed.
+     *
+     * @param semester semester number
+     * @param subject name of the subject
+     * @param mark mark for this subject (valid: "удовлетворительно", "хорошо", "отлично")
+     */
+    public void putGrade(int semester, String subject, String mark) throws IllegalArgumentException {
```

Comment:
```suggestion
    public void putGrade(int semester, String subject, String mark) {
```
IllegalArgumentException наследуется от RuntimeException, соответственно это непроверяемое исключение и его не обязательно декларировать в контракте метода